### PR TITLE
Kev/374_select

### PR DESCRIPTION
**Pull request by junhaow1**
# Pull Request Details

## What issue does this PR address

- ROLE: Allow the user to select multiple variables to set ROLE in one go #374

- Link to associated issue: #374 

## Checklist

Complete the check-list below to ensure your branch is ready for PR.

Flutter Style Guide: https://survivor.togaware.com/gnulinux/flutter-style.html

- [x] Screenshots included in linked issue
- [x] Changes adhere to the style and coding guideline
- [x] No confidential information
- [x] No duplicated content
- [x] No lint check errors related to your changes (`make prep` or `flutter analyze lib`)
- [ ] Pre-exisiting lint errors noted: [HERE]
- [ ] Integration test `make qtest.tmp` screenshot included in issue
- [x] Tested on device:
  - [ ] Linux
  - [ ] MacOS
  - [x] Windows
- [ ] Added two reviewers

## Finalising

Once PR discussion is complete and reviewers have approved:

- [ ] Merge dev into the branch
- [ ] Resolve any conflicts
- [ ] Add one line summary into CHANGELOG.md
- [ ] Bump appropriate version number in pubspec.yaml
- [ ] Push to git repository and review
- [ ] Merge PR into dev (gjwgit)

### CODE REVIEW

**Review by gjwgit — COMMENTED** [review]
- [x] Selecting the first row also selects the headline - it should not

image

File: lib/features/dataset/display.dart [hunk 1]
```diff
@@ -34,9 +35,12 @@ import 'package:flutter_riverpod/flutter_riverpod.dart';
 import 'package:rattle/constants/app.dart';
 import 'package:rattle/constants/markdown.dart';
 import 'package:rattle/constants/spacing.dart';
+import 'package:rattle/features/dataset/config.dart';
```
Comment (gjwgit) [inline lib/features/dataset/display.dart]: This should not be required for a single constant - the constant belongs in constants/spacing.dart - use one from there already

File: lib/features/dataset/display.dart [hunk 2]
```diff
@@ -34,9 +35,12 @@ import 'package:flutter_riverpod/flutter_riverpod.dart';
...
 import 'package:rattle/providers/meta_data.dart';
 import 'package:rattle/providers/page_controller.dart';
 import 'package:rattle/providers/path.dart';
+import 'package:rattle/providers/rolesTableRebuild.dart';
```
Comment (gjwgit) [inline lib/features/dataset/display.dart]: This is NOT the style of file naming we are using EVERY WHERE ELSE in this code - why have you suddenly changed the style?

File: lib/features/dataset/display.dart [hunk 3]
```diff
@@ -74,6 +78,10 @@ class _DatasetDisplayState extends ConsumerState<DatasetDisplay> {
   final int typeFlex = 4;
   final int contentFlex = 3;
 
+  // Track pressed keys for shift selection
+
+  bool _isShiftPressed = false;
```
Comment (gjwgit) [inline lib/features/dataset/display.dart:82]: Would the DataTable widget be better for supporting selecting rows in a standard way?

File: lib/features/dataset/display.dart [hunk 4]
```diff
@@ -141,16 +167,110 @@ class _DatasetDisplayState extends ConsumerState<DatasetDisplay> {
 
     updateVariablesProvider(ref);
 
+    Map<String, String> rolesOption = {
+      'IGNORE': ''' Ignore this dataset during analysis.
```
Comment (gjwgit) [inline lib/features/dataset/display.dart]: Where else in Rattle do we use all uppercase button labels? Is there a reason? Also, please use the same button style as ActivtiyButton so all buttons look the same. These standout as different to the other buttons in the app.

File: lib/providers/rolesTableRebuild.dart
Comment (gjwgit) [inline lib/providers/rolesTableRebuild.dart:1]: Rename file to standard

File: lib/features/dataset/display.dart (hunk 1 above)
Comment (junhaow1) [reply lib/features/dataset/display.dart]: done

File: lib/features/dataset/display.dart (hunk 2 above)
Comment (junhaow1) [reply lib/features/dataset/display.dart]: done

File: lib/providers/rolesTableRebuild.dart
Comment (junhaow1) [reply lib/providers/rolesTableRebuild.dart:1]: done

File: lib/features/dataset/display.dart (hunk 4 above)
Comment (junhaow1) [reply lib/features/dataset/display.dart]: done

File: lib/features/dataset/display.dart (hunk 3 above)
Comment (junhaow1) [reply lib/features/dataset/display.dart:82]: will solve this in https://github.com/gjwgit/rattleng/issues/581

**Comment by junhaow1** [comment]
ready to review again @gjwgit  :)

File: lib/constants/spacing.dart [hunk 5]
```diff
@@ -27,6 +27,10 @@ library;
 
 import 'package:gap/gap.dart';
 
+/// Space between widgets in a Row.
+
+const double widthSpace = 5;
```
Comment (gjwgit) [inline lib/constants/spacing.dart]: I am moving to using Gap(5).

You should be using one of the context dependent constants in this file, not the widthSpace literal which should be removed.

File: lib/providers/selectedRow.dart
Comment (gjwgit) [inline lib/providers/selectedRow.dart:1]: Filename ? We have a standard for a purpose - why keep flaunting it?

**Review by gjwgit — COMMENTED** [review]
FOR DISCUSSION IN STANDUP

**Pull request merged**